New inputs + controls

simurai opened the pull request:
This adds a few commonly used inputs and other controls:
- [x] `input-text`
- [x] `input-search`
- [x] `input-textarea`
- [x] `input-radio`
- [x] `input-checkbox`
- [x] `input-toggle`
- [x] `input-range`
- [x] `input-color`
- [x] `input-number`
- [x] `input-select`
- [x] `input-label`

![inputs](https://cloud.githubusercontent.com/assets/378023/16706747/c93b126e-45f2-11e6-9d89-aec748c163bf.gif)

Some things to decide:
- [x] Class names. Once it's out, it will have to stay till the next major version.
- [x] Sizes. I tried to use `em`s, but running into same [pixel snapping](https://github.com/atom/one-dark-ui/pull/144#issuecomment-227700428) issues. Might give up and just use fixed `px`s. :sob:

/cc @atom/design

simurai committed:
Add various input components

donokuda commented:
😍 😍 😍 

> Class names. Once it's out, it will have to stay till the next major version.

Class names look fine to me unless you have ran into issues with them clashing in other themes in the past.

> Sizes. I tried to use `em`s, but running into same pixel snapping issues. Might give up and just use fixed `px`s. 😭

No luck with your `scale()` approach? 😢

simurai commented:
> clashing in other themes in the past.

The `input-` prefix should guard a bit against clashes in other packages. It also hints that an `<input>` element + `type` should be used.

Just `select` might be a bit odd. We already have a `select-list`. So we could call it `select-menu`? Similar to a "context menu".

> No luck with your scale() approach?

That somewhat worked with that font icon, but normal elements seem even more aggressive and sometimes the snapping makes it look like this:

![screen shot 2016-07-07 at 12 08 34 pm](https://cloud.githubusercontent.com/assets/378023/16641470/c88bb6c4-443b-11e6-85de-8d479980c3a8.png)

Maybe keep `em`s for text related things like the text inputs, buttons, but `px` for radio/checkbox/toggle etc. that have no text.

Or try to use SVG, but then the markup gets complicated with wrapper elements and stuff. Custom elements would be neat: `<atom-radio>`, `<atom-checkbox>`. Maybe one day™

simurai committed:
Adjust sizes

- text inputs: `em`
- text-less controls `@component-icon-size`

simurai commented:
ok, it's now:

> Maybe keep `em`s for text related things like the text inputs, buttons, but `px` for radio/checkbox/toggle etc. that have no text.

![screen shot 2016-07-07 at 2 49 03 pm](https://cloud.githubusercontent.com/assets/378023/16644260/94a6f2b8-4456-11e6-8d34-392b03f1a14d.png)

thomasjo commented:
Can we do something about the "scroller" (aka. arrows) on the range input? Looks really out of place, and too small to be comfortable to click.

simurai committed:
Tweak “inactive” colors

simurai committed:
Style spin button

thomasjo commented on styles/inputs.less:
```diff
@@ -44,7 +44,19 @@
+    border-right: 4px solid @input-background-color; // a bit more padding
+    background-color: @input-background-color;
+  }
 }
```
Nice!

simurai commented:
Yeah, we could scale them up a bit:

![number](https://cloud.githubusercontent.com/assets/378023/16648776/9d2a7b5c-4470-11e6-9c26-b9778430bf30.gif)

But **pixelated**! :see_no_evil:

I replaced it with the `menulist` arrows. Only issue, there is no "active" state when clicking. Maybe still better:

![number](https://cloud.githubusercontent.com/assets/378023/16648672/24d327d0-4470-11e6-8bd2-6dbad986f01e.gif)

donokuda commented:
> I replaced it with the menulist arrows. Only issue, there is no "active" state when clicking. Maybe still better
> ![](https://cloud.githubusercontent.com/assets/378023/16648672/24d327d0-4470-11e6-8bd2-6dbad986f01e.gif)

I like this approach more despite the caveat that there aren't any affordances on how large the clickable area is (along with no active state.)

What are your thoughts on showing the `menulist` arrows on hover? That way it acts closer to how the arrows appear natively.

donokuda commented:
> Just select might be a bit odd. We already have a `select-list`. So we could call it `select-menu`? Similar to a "context menu".

👍 I like `select-menu` or perhaps `select-input` since it's a type of form input.

mikemcbride commented:
What about `input-select` to match the other input classes?

simurai committed:
Rename .select to .input-select

simurai commented:
> @mmcbride1007 What about `input-select` to match the other input classes?

Yeah, let's do that. Then it feels like it belongs together, also in the Styleguide.

simurai commented:
> @donokuda What are your thoughts on showing the menulist arrows on hover?

Ohh.. I just noticed, the arrows show while the input is focused, if not, it's on hover. So I guess good like that.

![number](https://cloud.githubusercontent.com/assets/378023/16674914/9086368e-44f7-11e6-8419-272242dc1854.gif)

simurai committed:
Add background to spin button

simurai commented:
Here another tweak with a background and some scaling when pressed to give a least a little confirmation.

![number](https://cloud.githubusercontent.com/assets/378023/16678820/a1f9b622-451d-11e6-9ef2-8018024f16f6.gif)

simurai committed:
Add label, legend and input sets

simurai commented:
Added `input-label`, `input-legend` and `input-set`. So it's easier to add groups.

![screen shot 2016-07-09 at 11 50 55 am](https://cloud.githubusercontent.com/assets/378023/16705664/76d51442-45cb-11e6-84b4-1df521e7ed32.png)

simurai committed:
:fire: Remove input-set + input-legend again

Let’s think about layout in another PR. This one just styles the components alone.

simurai committed:
Smaller radio dot

simurai committed:
Use mixins

simurai committed:
:art: Order classes

simurai committed:
Import inputs.less

simurai commented:
🚢

thomasjo commented:
Brilliant stuff! 😍
